nixos/lib/build-vms.nix

```diff
@@ -36,9 +36,15 @@ rec {
       baseModules =  (import ../modules/module-list.nix) ++
         [ ../modules/virtualisation/qemu-vm.nix
           ../modules/testing/test-instrumentation.nix # !!! should only get added for automated test runs
-          { key = "no-manual"; documentation.nixos.enable = false; }
-          { key = "qemu"; system.build.qemu = qemu; }
-          { key = "nodes"; _module.args.nodes = nodes; }
+          { key = "vm";
+            documentation.nixos.enable = false;
+            system.build.qemu = qemu;
+            _module.args.nodes = nodes;
+            # Ensure that NixOS configurations don't have -dev paths
+            # or gcc.out in their closures.
+            system.forbiddenDependencies = mkDefault "(-dev$)|(-gcc-[0-9\.]+$)|(gcc-wrapper)";
+            system.maxClosureSize = mkDefault (1024 * 1024 * 1024);
```

Comment:
Based on my experience with GHC, it will be the tests on ARM that run into such a check first.
Those are may be undermaintained, so they'd (wrongly) end up in the bag of tests considered flaky, so I wouldn't want to burden all tests with a (default) closure size check. Instead, we could check some key services or even just the empty NixOS on a _single_, popular `system`, such as `x86_64-linux`. (e.g. `nixosTests.empty-system-closure-size`, so the problem can be spotted at a glance)

Reply:
Sounds good, I think.